homeassistant/components/tado/coordinator.py

```diff
@@ -339,6 +339,17 @@ async def set_meter_reading(self, reading: int) -> dict[str, Any]:
         except RequestException as err:
             raise UpdateFailed(f"Error setting Tado meter reading: {err}") from err
 
+    async def set_child_lock(self, device_id: str, enabled: bool) -> None:
+        """Set child lock of device."""
+        try:
+            await self.hass.async_add_executor_job(
+                self._tado.set_child_lock,
+                device_id,
+                enabled,
+            )
+        except RequestException as exc:
+            raise UpdateFailed(f"Error setting Tado child lock: {exc}") from exc
```

Comment:
This is not an `UpdateFailed`, this rather is a `HomeAssistantError` as this is run in an action context

Reply:
Why is that? It updates the state remotely and also updates HA state afterwards. Plus, all the other methods in this api wrapper use this expection. Curious on why all the methods are fine using `UpdateFailed` but this particular one is not :)

Reply:
Everything used in an updating context `_async_update_data` should raise `UpdateFailed` when not working, but everything in an action context should raise `HomeAssistantError` if it is something happened during execution and `ServiceValidationError` if it was a user mistake

Reply:
So basically almost all the methods defined in this coordinator are subject to change?